workers/task-driver/src/driver.rs

```diff
@@ -397,14 +397,18 @@ impl TaskExecutor {
         notifications: TaskNotificationMap,
     ) -> Result<(), TaskDriverError> {
         // Create the task
-        let mut task = RunnableTask::<T>::from_descriptor(immediate, id, descriptor, ctx).await?;
-
-        // Run the task
-        let res = Self::run_task_to_completion(&mut task, args).await;
-
-        // Cleanup
-        let cleanup_res = task.cleanup(res.is_ok()).await;
-        let combined_res = res.and(cleanup_res);
+        let combined_res =
```

Comment:
disagree with this approach, let's validate this at the API layer. The whole point of this change is to give safety to events created through the API, this is too indirect of a way of returning an error to the API. Think about it from the client's perspective; the http task succeeds but they have to poll the task status to figure out whether their request was valid -- that's just a poor interface.

I'd suggest adding a safety check in the task descriptor so that the http call itself fails with `400` when we incorrectly reblind

Reply:
And when validating at the API layer we should use the back of queue wallet, it should be available already via a helper at the API layer

Reply:
The reason for this approach is to have certainty that we're checking a reblind progression on the exactly correct wallet, without having to make assumptions about the correctness of the wallet simulation functionality.

At the API layer we already apply reblinds on top of the back of queue wallet, so doing this check there would only tell us if we forgot to call `reblind` before invoking the descriptor constructor. However one of the issues we saw is that this did not accurately represent the state of the wallet after the queue had processed.

I agree that this is somewhat confusing from the client's perspective but it should be able to see that the task failed, this is something it has to handle regardless.

Reply:
talked off line, orienting this around preventing a footgun is more important than letting the client know early of a relayer mistake. approved